src/main/java/org/spongepowered/common/data/persistence/ConfigurateTranslator.java

```diff
@@ -75,64 +73,56 @@ private static DataContainer translateFromNode(ConfigurationNode node) {
         return dataContainer;
     }
 
-    private void translate(Map<Object, ? extends ConfigurationNode> node, DataView container, Map<?, ?> value) {
+    private void translate(Map<Object, ? extends ConfigurationNode> node, DataView container, Object key, Map<?, ?> value) {
         for (Map.Entry<?, ?> o : value.entrySet()) {
-            translate(node.get(o.getKey()), container);
+            ConfigurationNode n = node.get(o.getKey());
+            if (!n.getValueType().canHaveChildren()) {
+                container.set(of(String.valueOf(o.getKey())), n.getValue());
+            } else if (n.hasListChildren()) {
+                translate(n, container);
+            } else {
+                DataContainer sub = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
+                translate(n, sub);
+                container.set(of(String.valueOf(o.getKey())), sub);
+            }
         }
     }
 
-    private List<Object> translate(List<? extends ConfigurationNode> node, DataView container, List<?> value) {
+    private void translate(List<? extends ConfigurationNode> node, DataView container, Object key, List<?> value) {
         List<Object> list = new ArrayList<>(value.size());
         for (int i = 0; i < value.size(); i++) {
             ConfigurationNode n = node.get(i);
-            if (n.getValueType() == ValueType.SCALAR) {
+            if (!n.getValueType().canHaveChildren()) {
                 list.add(n.getValue());
-            } else if (n.hasMapChildren()) {
-                DataContainer clean = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
-                translate(n.getChildrenMap(), clean, (Map<?, ?>) n.getValue());
-                list.add(clean);
             } else {
-                DataContainer clean = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
-                translate(n, clean);
-                list.add(clean);
+                DataContainer sub = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
+                translate(n, sub);
+                list.add(sub);
             }
         }
-        return list;
+        container.set(of(String.valueOf(key)), list);
     }
 
     private void translate(ConfigurationNode node, DataView container) {
         Object key = node.getKey();
         Object value = node.getValue();
         if (node.hasMapChildren()) {
```

Comment:
`if/else` for map, list, and normal value. Individual `if` conditions and `return` is complicating things here.